google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -252,18 +275,20 @@ public int hashCode() {
   }
 
   Query(FirestoreImpl firestore, ResourcePath path) {
-    this(firestore, path, new QueryOptions());
+    this.firestore = firestore;
+    this.options = new QueryOptions(path.getParent(), path.getId());
   }
 
-  protected Query(
-      FirestoreImpl firestore,
-      ResourcePath path,
-      QueryOptions queryOptions) { // Elevated access level for mocking.
-    Preconditions.checkArgument(
-        path.isCollection(), "Invalid path specified. Path should point to a collection");
+  Query(FirestoreImpl firestore, String collectionId) {
+    QueryOptions queryOptions = new QueryOptions(firestore.getResourcePath(), collectionId);
+    queryOptions.allDescendants = true;
 
     this.firestore = firestore;
-    this.path = path;
+    this.options = queryOptions;
```

Comment:
I made this change, which is now possible thanks to the QueryOptions.Builder. Without it, I had no way to initialize `allDescendants`.

I opted not to make these factory methods since CollectionReference inherits from Query and I like the consistency of having two constructors rather than one constructor and one factory method.